### PR TITLE
New docs page: Incoming Webhooks

### PR DESCRIPTION
**Pull request by karensawrey**
Not quite happy with the intro, suggestions are mighty welcome.

### CODE REVIEW

File: pages/pipelines/incoming_webhooks.md.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Incoming Webhooks
+
+Incoming webhooks are sent to Buildkite by source control providers ([GitHub](/docs/integrations/github), [GitLab](/docs/integrations/gitlab), [Bitbucket](/docs/integrations/bitbucket), etc.) to trigger builds. This page covers the most frequent questions that arise regarding the security of incoming webhooks in Buildkite.
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:3]:
```suggestion
Incoming webhooks are sent to Buildkite by source control providers ([GitHub](/docs/integrations/github), [GitLab](/docs/integrations/gitlab), [Bitbucket](/docs/integrations/bitbucket), etc.) to trigger builds. This page answers the most frequent questions about the security of incoming webhooks in Buildkite.    
```

File: pages/pipelines/incoming_webhooks.md.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Incoming Webhooks
+
+Incoming webhooks are sent to Buildkite by source control providers ([GitHub](/docs/integrations/github), [GitLab](/docs/integrations/gitlab), [Bitbucket](/docs/integrations/bitbucket), etc.) to trigger builds. This page covers the most frequent questions that arise regarding the security of incoming webhooks in Buildkite.    
+
+### What kind of information on incoming webhooks is logged by Buildkite? 
+Buildkite only logs and temporarily stores the incoming webhook information as it was received, the relevant HTTP headers, and the remote IP for diagnostics purposes. This information is soon erased and cannot be used for audit.
+ 
+### How long does Buildkite store the information regarding incoming webhooks?
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:8]:
```suggestion
### How long does Buildkite store the information on incoming webhooks?
```

File: pages/pipelines/incoming_webhooks.md.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Buildkite stores the information on recent incoming webhooks received per pipeline for approximately a week. It is done for troubleshooting purposes only — for example, to diagnose why a given hook didn’t trigger a build. This functionality is not exposed in the UI.
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:9]:
```suggestion
Buildkite stores the information on recent incoming webhooks received per pipeline for approximately one week. It is used for troubleshooting purposes only — for example, to diagnose why a given hook didn’t trigger a build. This functionality is not exposed in the UI.
```

File: pages/pipelines/incoming_webhooks.md.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ 
+### What security information is stored in the secret in a Buildkite URL? Does it only say that I — as a user — am talking to Buildkite, and not that the webhook is triggered from my GitHub organization (or any other GitHub entity)?
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:11]:
```suggestion
### What security information is stored in the secret in a Buildkite URL?
```

looks better? What do you think?

File: pages/pipelines/incoming_webhooks.md.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ 
+### How long does Buildkite store the information regarding incoming webhooks?
+Buildkite stores the information on recent incoming webhooks received per pipeline for approximately a week. It is done for troubleshooting purposes only — for example, to diagnose why a given hook didn’t trigger a build. This functionality is not exposed in the UI.
+ 
+### What security information is stored in the secret in a Buildkite URL? Does it only say that I — as a user — am talking to Buildkite, and not that the webhook is triggered from my GitHub organization (or any other GitHub entity)? 
+Secrets are pipeline-specific. A secret can’t prove that it hasn’t been acquired by somebody else and added as a webhook inside their own GitHub repository (which is equally true of any other secret). However, all incoming webhooks coming from GitHub-hosted repositories are IP-filtered to provide an additional layer of security.
+
+As a Buildkite user, you can [setup a pipeline and request a URL](/docs/integrations/github#set-up-a-new-pipeline-for-a-github-repository) to add it to your GitHub organization as a webhook. It means that you would prove that it is coming from your GitHub organization.
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:14]:
```suggestion
As a Buildkite user, you can [setup a pipeline and request a URL](/docs/integrations/github#set-up-a-new-pipeline-for-a-github-repository) to add it to your GitHub organization as a webhook which would prove that it is coming from your GitHub organization.
```

**Review by plaindocs — APPROVED** [review]
This great to me. A few minor suggestions inline.

@yob any thing to add?

File: pages/pipelines/incoming_webhooks.md.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Buildkite only logs and temporarily stores the incoming webhook information as it was received, the relevant HTTP headers, and the remote IP for diagnostics purposes. This information is soon erased and cannot be used for audit.
+ 
+### How long does Buildkite store the information regarding incoming webhooks?
+Buildkite stores the information on recent incoming webhooks received per pipeline for approximately a week. It is done for troubleshooting purposes only — for example, to diagnose why a given hook didn’t trigger a build. This functionality is not exposed in the UI.
+ 
+### What security information is stored in the secret in a Buildkite URL? Does it only say that I — as a user — am talking to Buildkite, and not that the webhook is triggered from my GitHub organization (or any other GitHub entity)? 
+Secrets are pipeline-specific. A secret can’t prove that it hasn’t been acquired by somebody else and added as a webhook inside their own GitHub repository (which is equally true of any other secret). However, all incoming webhooks coming from GitHub-hosted repositories are IP-filtered to provide an additional layer of security.
```
Comment (yob) [inline pages/pipelines/incoming_webhooks.md.erb:12]: Do we want to include IP filtering behaviour for other providers a well?

For BitBucket Server, GitHub Enterprise Server, GitLab Community and Enterprise: customers can provide permitted IPs in their organization settings


For BitBucket standard, there's currently no IP filtering for webhooks.

**Review by yob — COMMENTED** [review]
This matches my technical understanding of the way incoming webhooks are handled :+1:

File: pages/pipelines/incoming_webhooks.md.erb (hunk 4 above)
Comment (karensawrey) [reply pages/pipelines/incoming_webhooks.md.erb:11]: I need to think about it a bit.

File: pages/pipelines/incoming_webhooks.md.erb (hunk 6 above)
Comment (karensawrey) [reply pages/pipelines/incoming_webhooks.md.erb:12]: Thanks for pointing this out, @yob ! I'll add this information to the document.

File: pages/pipelines/incoming_webhooks.md.erb (hunk 4 above)
Comment (karensawrey) [reply pages/pipelines/incoming_webhooks.md.erb:11]: I'm not totally happy with just cutting out the rest of the question here, but I cannot quite rephrase the long question at the moment either. If the muse does visit me regarding this entry, I'll submit an extended version as a fix.

**Comment by plaindocs** [comment]
:shipit:

File: pages/pipelines/incoming_webhooks.md.erb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ 
+### How long does Buildkite store the information on incoming webhooks?
+Buildkite stores the information on recent incoming webhooks received per pipeline for approximately one week. It is used for troubleshooting purposes only — for example, to diagnose why a given hook didn’t trigger a build. This functionality is not exposed in the UI.
+ 
+### What security information is stored in the secret in a Buildkite URL?
+Secrets are pipeline-specific. A secret can’t prove that it hasn’t been acquired by somebody else and added as a webhook inside their own GitHub repository (which is equally true of any other secret). However, all incoming webhooks coming from GitHub-hosted repositories are IP-filtered to provide an additional layer of security. As a Buildkite user, you can [setup a pipeline and request a URL](/docs/integrations/github#set-up-a-new-pipeline-for-a-github-repository) to add it to your GitHub organization as a webhook. It means that you would prove that it is coming from your GitHub organization.
```
Comment (plaindocs) [inline pages/pipelines/incoming_webhooks.md.erb:12]:
```suggestion
Secrets are pipeline-specific. A secret can’t prove that it hasn’t been acquired by somebody else and added as a webhook inside their own GitHub repository (which is equally true of any other secret). However, all incoming webhooks coming from GitHub-hosted repositories are IP-filtered to provide an additional layer of security. As a Buildkite user, you can [set up a pipeline and request a URL](/docs/integrations/github#set-up-a-new-pipeline-for-a-github-repository) to add it to your GitHub organization as a webhook. It means that you would prove that it is coming from your GitHub organization.   
```

**Comment by keithpitt** [comment]
Great doc!

Oh! And a good thing to perhaps add is: "can the webhook that triggered a build be accessed in the UI or API?"

The answer is: yes! Ping me and I'll give you the details.

**Pull request merged**